plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/labeling/helper/StyledTextCalculationHelper.xtend

```diff
@@ -216,7 +224,12 @@ class StyledTextCalculationHelper {
 	 */
 	def dispatch StyledString dispatchGetStyledText(TField tfield) {
 		var styledText = getLabelProvider.getSuperStyledText(tfield);
-		if (tfield.typeRef !== null && styledText !== null) {
+		if (styledText === null) return unknown;
+		styledText.setStyle(0, styledText.length, N4JSStylers.FIELD_OR_VAR_STYLER)
+		if (tfield.isOptional) {
+			styledText.append("?");
```

Comment:
I did that on purpose: the "?" is not part of the name. Makes it slightly better to read (I tested that myself ;-) ). Actually I introduce the field colors after finding myself looking for the fields in longer lists.